src/routes/files/handlers.js

```diff
@@ -0,0 +1,103 @@
+import { v4 as uuidv4 } from 'uuid';
+import handleErrors from '../../lib/apiErrorHandler';
+import { sequelize, File } from '../../models';
+import s3Uploader from '../../lib/s3Uploader';
+
+const fs = require('fs');
+const fileType = require('file-type');
+const multiparty = require('multiparty');
+
+const namespace = 'SERVICE:FILES';
+
+const ATTACHMENT = 'ATTACHMENT';
+const RESOURCE = 'RESOURCE';
+
+const logContext = {
+  namespace,
+};
+
+export const createFileMetaData = async (
+  originalFileName,
+  s3FileName,
+  reportId,
+  attachmentType,
+) => {
+  const newFile = {
+    activityReportId: reportId,
+    originalFileName,
+    attachmentType,
+    key: s3FileName,
+    status: 'UPLOADING',
+  };
+  let file;
+  try {
+    await sequelize.transaction(async (transaction) => {
+      file = await File.create(newFile, transaction);
+    });
+    return file.dataValues;
+  } catch (error) {
+    return error;
+  }
+};
+
+export const updateStatus = async (fileId, fileStatus) => {
+  let file;
+  try {
+    await sequelize.transaction(async (transaction) => {
+      file = await File.update({ status: fileStatus }, { where: { id: fileId } }, transaction);
+    });
+    return file.dataValues;
+  } catch (error) {
+    return error;
+  }
+};
+
+export default async function uploadHandler(req, res) {
+  const form = new multiparty.Form();
+  form.parse(req, async (error, fields, files) => {
+    if (error) {
+      res.status(500).send(error);
+    }
+    let buffer;
+    let metadata;
+    let fileName;
+    let type;
+
+    try {
+      if (!files.File) {
+        res.status(400).send({ error: 'file required' });
+        return;
+      }
+      const { path, originalFilename } = files.File[0];
+      const { reportId, attachmentType } = fields;
+      if (!reportId) {
+        res.status(400).send({ error: 'requestId required' });
+        return;
+      }
+      if (!attachmentType) {
+        res.status(400).send({ error: 'attachmentType required' });
+        return;
+      }
+      if (attachmentType[0] !== ATTACHMENT && attachmentType[0] !== RESOURCE) {
+        res.status(400).send({ error: `incorrect attachmentType. Wanted: ${ATTACHMENT} or ${RESOURCE}. Got: ${attachmentType[0]}` });
+        return;
+      }
+      buffer = fs.readFileSync(path);
+      type = await fileType.fromFile(path);
+      fileName = `${uuidv4()}.${type.ext}`;
+      metadata = await createFileMetaData(originalFilename, fileName, reportId, attachmentType[0]);
+    } catch (err) {
+      await handleErrors(req, res, err, logContext);
```

Comment:
I think we need to exit from the function here. If we encounter an error above it will be handled here and continue on trying to upload the file to s3 on line 93.

Reply:
Another option would be to move the additional `try catch` into this upper `try catch` but that may have additional consequences I haven't noticed. If everything was in a try catch I can see how you could have a save the file in the db and update the status in a single transaction, uploading the file to s3 in between.

Reply:
I think exiting works. If we can't create metadata in the DB, we shouldn't continue with the upload.